actions.js

```diff
@@ -110,7 +110,7 @@ const actions = {
     }
 
     do {
-      const min_action = (function() {
```

Comment:
I had set this as a const because you're only changing the object properties which doesn't require reassignment.  This comment applies for pretty much the rest of the consts where you only manipulate the object properties.

Reply:
In this particular case it could be const, will revert.